src/scene/materials/standard-material-options-builder.js

```diff
@@ -308,42 +308,34 @@ StandardMaterialOptionsBuilder.prototype._collectLights = function (lType, light
     }
 };
 
+const arraysEqual = (a, b) => {
```

Comment:
see set-utils.js. Maybe we need array-utils.js for this one.

Reply:
TBH I'd love to add a general 'deepCompare' function which handles comparing objects, arrays and simple data types.